src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -31,7 +31,10 @@ internal interface IWorkspaceProjectContext : IDisposable
         ProjectId Id { get; }
 
         // Options.
+
+        [Obsolete("To avoid contributing to the large object heap, use SetOptions(IEnumerable<string>). This API will be removed in the future.")]
         void SetOptions(string commandLineForOptions);
+        void SetOptions(IEnumerable<string> arguments);
```

Comment:
Can we make this an ImmutableArray so we don't have any concerns about multiple enumeration or something like that?